### PR TITLE
FFmpeg producer allways resample to 48kHz

### PR DESCRIPTION
**Pull request by ryci-us**
Fixes audio out of sync when source is not in 48kHz sample rate.
Fixes: https://github.com/CasparCG/server/issues/1194 https://github.com/CasparCG/server/issues/1170

### CODE REVIEW

File: src/modules/ffmpeg/producer/av_producer.cpp [hunk 1]
```diff
@@ -223,7 +223,7 @@ struct Filter
                 filter_spec = "anull";
             }
 
-            filter_spec += (boost::format(",aresample=async=1000:first_pts=%d:min_comp=0.01,asetrate=r=%d,"
+            filter_spec += (boost::format(",aresample=async=1000:first_pts=%d:min_comp=0.01,aresample=%d,"
```
Comment (premultiply) [inline src/modules/ffmpeg/producer/av_producer.cpp]: Looks to me as this PR uses resample filter twice now.
Can this be combined into only one filter step?

Maybe
`aresample=osr=48000:async=1000:first_pts=%d:min_comp=0.01`
would be a better solution?

Comment (ryci-us) [reply src/modules/ffmpeg/producer/av_producer.cpp]: i think no, first filter Stretch/squeeze the data to make it match the timestamps, second one changes sample rate to 48kHz.

Comment (ryci-us) [reply src/modules/ffmpeg/producer/av_producer.cpp]: as i remember async is not a parameter, it's another filter which computes new sample rate to match sync, and its results are used to adjust aresample output sample rate.

Comment (ronag) [reply src/modules/ffmpeg/producer/av_producer.cpp]: You can combine

Comment (premultiply) [reply src/modules/ffmpeg/producer/av_producer.cpp]: async is a parameter of the aresample-filter

Comment (ryci-us) [reply src/modules/ffmpeg/producer/av_producer.cpp]: yes, you are right. Everyhing works. I will make new PR to fix it.

**Pull request merged**